server/sonar-server/src/main/java/org/sonar/server/webhook/WebHooks.java

```diff
@@ -24,24 +24,32 @@
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 import org.sonar.api.config.Configuration;
+import org.sonar.db.component.ComponentDto;
 
 import static java.util.Objects.requireNonNull;
 
 public interface WebHooks {
+
   /**
-   * Tells whether any webHook is configured at all for the specified {@link Configuration}.
+   * Tells whether any webHook is configured for the specified {@link Configuration}.
    *
    * <p>
    * This can be used to not do consuming operations before calling
-   * {@link #sendProjectAnalysisUpdate(Configuration, Analysis, Supplier)}
+   * {@link #sendProjectAnalysisUpdate(ComponentDto, Analysis, Supplier)}
+   */
+  boolean isEnabled(ComponentDto projectDto);
+
+  /**
+   * Calls all WebHooks configured in the specified {@link Configuration} for the specified analysis with the
+   * {@link WebhookPayload} provided by the specified Supplier.
    */
-  boolean isEnabled(Configuration configuration);
+  void sendProjectAnalysisUpdate(ComponentDto projectDto, Analysis analysis, Supplier<WebhookPayload> payloadSupplier);
 
   /**
    * Calls all WebHooks configured in the specified {@link Configuration} for the specified analysis with the
    * {@link WebhookPayload} provided by the specified Supplier.
    */
-  void sendProjectAnalysisUpdate(Configuration configuration, Analysis analysis, Supplier<WebhookPayload> payloadSupplier);
+  void sendProjectAnalysisUpdate(Analysis analysis, Supplier<WebhookPayload> payloadSupplier);
```

Comment:
I don't understand why this method is not fully replaced by the first one at line 46. ComponentDto is mandatory on all cases, isn't it ?

Reply:
yep, i have moved the fetch of the componentDto one level up in order to keep a single method

Reply:
indeed but not present in the upper context of the compute engine, i wanted to avoid spreading this dependency..